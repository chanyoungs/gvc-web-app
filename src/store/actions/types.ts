import { ExtendedFirestoreInstance, getFirebase } from "react-redux-firebase"
import { ThunkAction } from "redux-thunk"
import { Font } from "src/components/Level1/Dialogs/FontDialog"
import { IBibleRef } from "src/components/Pages/BiblePage"

import { IFBError, INotice, IReport, IReports, TPrayerQueries } from "../../types"
import { AppState } from "../reducers/rootReducer"

// Setup
export type ThunkActionCustom<ReturnType> = ThunkAction<
  ReturnType,
  AppState,
  {
    getFirebase: typeof getFirebase
    // getFirebase: (() => typeof firebase) & typeof getFirebase
    getFirestore: () => ExtendedFirestoreInstance
  },
  AppActions
>

// ServiceWorker Actions
export const SERVICE_WORKER_INIT = "SERVICE_WORKER_INIT"
export interface ServiceWorkerInit {
  type: typeof SERVICE_WORKER_INIT
  payload: boolean
}

export const SERVICE_WORKER_UPDATE = "SERVICE_WORKER_UPDATE"
export interface ServiceWorkerUpdate {
  type: typeof SERVICE_WORKER_UPDATE
  payload: ServiceWorkerRegistration
}

export type ServiceWorkerTypes = ServiceWorkerInit | ServiceWorkerUpdate

// Report actions
export const UPDATE_REPORT = "UPDATE_REPORT"
export interface UpdateReport {
  type: typeof UPDATE_REPORT
  payload: IReport
}

export const UPDATE_BATCH_REPORTS = "UPDATE_BATCH_REPORTS"
export interface UpdateBatchReports {
  type: typeof UPDATE_BATCH_REPORTS
  payload: IReports
}

export const UPDATE_BATCH_REPORTS_ERROR = "UPDATE_BATCH_REPORTS_ERROR"
export interface UpdateBatchReportsError {
  type: typeof UPDATE_BATCH_REPORTS_ERROR
  payload: IFBError
}

export const UPDATE_REPORT_ERROR = "UPDATE_REPORT_ERROR"
export interface UpdateReportError {
  type: typeof UPDATE_REPORT_ERROR
  payload: IFBError
}

export const UPDATE_PRAYER = "UPDATE_PRAYER"
export interface UpdatePrayer {
  type: typeof UPDATE_PRAYER
  payload: { reportId: string; prayer: string }
}

export const UPDATE_PRAYER_ERROR = "UPDATE_PRAYER_ERROR"
export interface UpdatePrayerError {
  type: typeof UPDATE_PRAYER_ERROR
  payload: IFBError
}

export const UPDATE_ATTENDANCE = "UPDATE_ATTENDANCE"
export interface UpdateAttendance {
  type: typeof UPDATE_ATTENDANCE
  payload: { reportId: string; attendance: IReport["attendance"] }
}

export const UPDATE_ATTENDANCE_ERROR = "UPDATE_ATTENDANCE_ERROR"
export interface UpdateAttendanceError {
  type: typeof UPDATE_ATTENDANCE_ERROR
  payload: IFBError
}

export type ReportActionTypes =
  | UpdateReport
  | UpdateBatchReports
  | UpdateBatchReportsError
  | UpdateReportError
  | UpdatePrayer
  | UpdatePrayerError
  | UpdateAttendance
  | UpdateAttendanceError

// Alert actions
export const ALERT_SAVED = "ALERT_SAVED"
export interface AlertSaved {
  type: typeof ALERT_SAVED
  payload: boolean
}

export const ALERT_SAVED_ERROR = "ALERT_SAVED_ERROR"
export interface AlertSavedError {
  type: typeof ALERT_SAVED_ERROR
  payload: IFBError
}

export type AlertActionTypes = AlertSaved | AlertSavedError

// Theme actions
export const UPLOAD_THEME = "UPLOAD_THEME"
export interface UploadTheme {
  type: typeof UPLOAD_THEME
}

export const UPLOAD_THEME_ERROR = "UPLOAD_THEME_ERROR"
export interface UploadThemeError {
  type: typeof UPLOAD_THEME_ERROR
  payload: IFBError
}

export const DELETE_THEME = "DELETE_THEME"
export interface DeleteTheme {
  type: typeof DELETE_THEME
}

export const DELETE_THEME_ERROR = "DELETE_THEME_ERROR"
export interface DeleteThemeError {
  type: typeof DELETE_THEME_ERROR
  payload: IFBError
}

export const SET_CURRENT_THEME_NAME = "SET_CURRENT_THEME_NAME"
export interface SetCurrentThemeName {
  type: typeof SET_CURRENT_THEME_NAME
  payload: string
}

export const SET_CURRENT_THEME_NAME_ERROR = "SET_CURRENT_THEME_NAME_ERROR"
export interface SetCurrentThemeNameError {
  type: typeof SET_CURRENT_THEME_NAME_ERROR
  payload: IFBError
}

export type ThemeActionTypes =
  | UploadTheme
  | UploadThemeError
  | SetCurrentThemeName
  | SetCurrentThemeNameError
  | DeleteTheme
  | DeleteThemeError

// Theme actions
export const UPLOAD_FONT = "UPLOAD_FONT"
export interface UploadFont {
  type: typeof UPLOAD_FONT
}

export const UPLOAD_FONT_ERROR = "UPLOAD_FONT_ERROR"
export interface UploadFontError {
  type: typeof UPLOAD_FONT_ERROR
  payload: IFBError
}

export const DELETE_FONT = "DELETE_FONT"
export interface DeleteFont {
  type: typeof DELETE_FONT
}

export const DELETE_FONT_ERROR = "DELETE_FONT_ERROR"
export interface DeleteFontError {
  type: typeof DELETE_FONT_ERROR
  payload: IFBError
}

export type FontActionTypes =
  | UploadFont
  | UploadFontError
  | DeleteFont
  | DeleteFontError

// Notice actions
export const CREATE_NOTICE = "CREATE_NOTICE"
export interface CreateNotice {
  type: typeof CREATE_NOTICE
  payload: INotice
}

export const CREATE_NOTICE_ERROR = "CREATE_NOTICE_ERROR"
export interface CreateNoticeError {
  type: typeof CREATE_NOTICE_ERROR
  payload: Error
}

export const DELETE_NOTICE = "DELETE_NOTICE"
export interface DeleteNotice {
  type: typeof DELETE_NOTICE
  payload: string
}

export const DELETE_NOTICE_ERROR = "DELETE_NOTICE_ERROR"
export interface DeleteNoticeError {
  type: typeof DELETE_NOTICE_ERROR
  payload: IFBError
}

export type NoticeActionTypes =
  | CreateNotice
  | CreateNoticeError
  | DeleteNotice
  | DeleteNoticeError

// Prayer actions
export const QUERY_PRAYER = "QUERY_PRAYER"
export interface QueryPrayer {
  type: typeof QUERY_PRAYER
  payload: TPrayerQueries
}

export const QUERY_PRAYER_ERROR = "QUERY_PRAYER_ERROR"
export interface QueryPrayerError {
  type: typeof QUERY_PRAYER_ERROR
  payload: IFBError
}

export type PrayerActionTypes = QueryPrayer | QueryPrayerError

// Auth actions
export const SIGN_UP = "SIGN_UP"
export interface SignUp {
  type: typeof SIGN_UP
}

export const SIGN_UP_ERROR = "SIGN_UP_ERROR"
export interface SignUpError {
  type: typeof SIGN_UP_ERROR
  payload: IFBError
}

export const MEMBER_PROFILE_CREATED = "MEMBER_PROFILE_CREATED"
export interface MemberProfileCreated {
  type: typeof MEMBER_PROFILE_CREATED
}

export const MEMBER_PROFILE_CREATED_ERROR = "MEMBER_PROFILE_CREATED_ERROR"
export interface MemberProfileCreatedError {
  type: typeof MEMBER_PROFILE_CREATED_ERROR
  payload: IFBError
}

export const SIGN_IN = "SIGN_IN"
export interface SignIn {
  type: typeof SIGN_IN
}

export const SIGN_IN_ERROR = "SIGN_IN_ERROR"
export interface SignInError {
  type: typeof SIGN_IN_ERROR
  payload: IFBError
}

export const REMEMBER_ME = "REMEMBER_ME"
export interface RememberMe {
  type: typeof REMEMBER_ME
  payload: boolean
}

export const REMEMBER_ME_ERROR = "REMEMBER_ME_ERROR"
export interface RememberMeError {
  type: typeof REMEMBER_ME_ERROR
  payload: IFBError
}

export const RESET_PASSWORD = "RESET_PASSWORD"
export interface ResetPassword {
  type: typeof RESET_PASSWORD
}

export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR"
export interface ResetPasswordError {
  type: typeof RESET_PASSWORD_ERROR
  payload: IFBError
}

export const SIGN_OUT = "SIGN_OUT"
export interface SignOut {
  type: typeof SIGN_OUT
}

export const SIGN_OUT_ERROR = "SIGN_OUT_ERROR"
export interface SignOutError {
  type: typeof SIGN_OUT_ERROR
  payload: IFBError
}

export const EDIT_PROFILE = "EDIT_PROFILE"
export interface EditProfile {
  type: typeof EDIT_PROFILE
}

export const EDIT_PROFILE_ERROR = "EDIT_PROFILE_ERROR"
export interface EditProfileError {
  type: typeof EDIT_PROFILE_ERROR
  payload: IFBError
}

export const UPLOAD_PHOTO = "UPLOAD_PHOTO"
export interface UploadPhoto {
  type: typeof UPLOAD_PHOTO
}

export const UPLOAD_PHOTO_ERROR = "UPLOAD_PHOTO_ERROR"
export interface UploadPhotoError {
  type: typeof UPLOAD_PHOTO_ERROR
  payload: Error
}

export const DELETE_PHOTO = "DELETE_PHOTO"
export interface DeletePhoto {
  type: typeof DELETE_PHOTO
}

export const DELETE_PHOTO_ERROR = "DELETE_PHOTO_ERROR"
export interface DeletePhotoError {
  type: typeof DELETE_PHOTO_ERROR
  payload: IFBError
}

export type AuthActionTypes =
  | SignUp
  | SignUpError
  | MemberProfileCreated
  | MemberProfileCreatedError
  | SignIn
  | SignInError
  | RememberMe
  | RememberMeError
  | ResetPassword
  | ResetPasswordError
  | SignOut
  | SignOutError
  | EditProfile
  | EditProfileError
  | UploadPhoto
  | UploadPhotoError
  | DeletePhoto
  | DeletePhotoError
  | DeleteNotice
  | DeleteNoticeError
  | UploadTheme
  | UploadThemeError
  | SetCurrentThemeName
  | SetCurrentThemeNameError
  | DeleteTheme
  | DeleteThemeError

// AppBar Actions
export const SEARCH_ON_CHANGE = "SEARCH_ON_CHANGE"
export interface SearchOnChange {
  type: typeof SEARCH_ON_CHANGE
  payload: string
}

export const SET_DRAWER_OPEN = "SET_DRAWER_OPEN"
export interface SetDrawerOpen {
  type: typeof SET_DRAWER_OPEN
  payload: boolean
}

export type AppBarActionTypes = SearchOnChange | SetDrawerOpen

// Bible Actions
export const SET_BIBLE_REFERENCE = "SET_BIBLE_REFERENCE"
export interface SetBibleReference {
  type: typeof SET_BIBLE_REFERENCE
  payload: IBibleRef
}

export const SET_BIBLE_REFERENCE_ERROR = "SET_BIBLE_REFERENCE_ERROR"
export interface SetBibleReferenceError {
  type: typeof SET_BIBLE_REFERENCE_ERROR
  payload: IFBError
}

export type BibleActionTypes = SetBibleReference | SetBibleReferenceError

// All Action types
export type AppActions =
  | NoticeActionTypes
  | PrayerActionTypes
  | AuthActionTypes
  | AppBarActionTypes
  | BibleActionTypes
  | ThemeActionTypes
  | FontActionTypes
  | AlertActionTypes
  | ReportActionTypes
  | ServiceWorkerTypes
