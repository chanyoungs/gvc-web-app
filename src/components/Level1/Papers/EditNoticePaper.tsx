import Button from "@material-ui/core/Button"
import Paper from "@material-ui/core/Paper"
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles"
import TextField from "@material-ui/core/TextField"
import React, { FC } from "react"
import { useDispatch } from "react-redux"

import { updateNotice } from "../../../store/actions/noticeActions"
import { INotice, INoticeWithMeta } from "../../../types"

export interface EditNoticePaperProps {
  notice: INoticeWithMeta
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    paper: {
      background: theme.palette.grey[700],
      padding: theme.spacing(2),
      display: "flex",
      overflow: "auto",
      flexDirection: "column",
      height: "100%",
      margin: "10px",
    },
  })
)

export const EditNoticePaper: FC<EditNoticePaperProps> = ({ notice }) => {
  const classes = useStyles()
  const [values, setValues] = React.useState<INotice>({
    title: notice.title,
    content: notice.content,
  })

  React.useEffect(() => {
    setValues(notice)
  }, [notice])

  const onChange = (name: keyof INotice) => (
    e: React.ChangeEvent<HTMLInputElement>
  ) => {
    setValues({ ...values, [name]: e.target.value })
  }

  const dispatch = useDispatch()
  const onSubmit = (values: INoticeWithMeta) => (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    e.preventDefault()
    setValues({
      title: "",
      content: "",
    })
    dispatch(updateNotice(values))
  }

  return (
    <Paper className={classes.paper}>
      <div data-testid="component-edit-noticepaper">
        <TextField
          data-testid="title"
          label="Title"
          fullWidth
          onChange={onChange("title")}
          value={values.title}
        />
        <TextField
          data-testid="content"
          id="standard-multiline-flexible"
          fullWidth
          multiline
          rowsMax="4"
          label="Content"
          onChange={onChange("content")}
          value={values.content}
        />
        <Button
          data-testid="submit"
          onClick={onSubmit({
            ...values,
            id: notice.id,
            createdAt: notice.createdAt,
          })}
        >
          Update notice
        </Button>
      </div>
    </Paper>
  )
}
