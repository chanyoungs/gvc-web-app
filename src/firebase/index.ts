import "firebase/auth"

import * as firebase from "firebase/app"
import "firebase/firestore"

var config = {
  apiKey: "AIzaSyBQf0kjL5VgL9IJuEtqWHevRkOMTBmVwiE",
  authDomain: "london-gvc.firebaseapp.com",
  databaseURL: "https://london-gvc.firebaseio.com",
  projectId: "london-gvc",
  storageBucket: "london-gvc.appspot.com",
  messagingSenderId: "139635867699",
  appId: "1:139635867699:web:649f0b856111fee7"
}

export const app = firebase.initializeApp(config)
export const auth = firebase.auth
export const db = firebase.firestore()