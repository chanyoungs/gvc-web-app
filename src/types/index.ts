// ---Notices---
export interface INotice {
  title: string
  content: string
}

export interface INoticeWithMeta extends INotice {
  id: string
  createdAt: Date
}

// ---Auth---
// Elements
export interface IEmail {
  email: string
}
export interface ISetSubmitting {
  setSubmitting: (isSubmitting: boolean) => void
}
export interface IOpenAlert {
  openAlert: () => void
}
export interface ICredentials extends IEmail {
  password: string
}

// Auth Interface
export interface IResetPassword extends IEmail, ISetSubmitting, IOpenAlert {}
export interface ISignIn extends ICredentials, ISetSubmitting {
  rememberMe: boolean
  redirectOnSignIn: () => void
}
export interface ISignUp extends ICredentials, ISetSubmitting, IOpenAlert {
  name: string
  dob: Date | null
  agreeTAndC: boolean
}

// Firebase Error Interface
export interface IFBError {
  code: string
  message: string
}

// ---Pages---
export type Paths =
  | "/"
  | "/myaccount"
  | "/prayers"
  | "/members"
  | "/calendar"
  | "/public"
  | "/private"
