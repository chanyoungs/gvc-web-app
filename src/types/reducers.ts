// import { IFBError } from "."

// export interface AuthState {
//   signInError: IFBError | null
//   signUpError: IFBError | null
// }
