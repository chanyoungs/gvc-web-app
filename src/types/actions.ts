import { ExtendedFirestoreInstance, getFirebase } from "react-redux-firebase"
import { ThunkAction } from "redux-thunk"

import { IFBError, INotice } from "."
import { AppState } from "../store/reducers/rootReducer"

// Setup
export type ThunkActionCustom<ReturnType> = ThunkAction<
  ReturnType,
  AppState,
  {
    getFirebase: typeof getFirebase
    // getFirebase: (() => typeof firebase) & typeof getFirebase
    getFirestore: () => ExtendedFirestoreInstance
  },
  AppActions
>

// Notice actions
export const CREATE_NOTICE = "CREATE_NOTICE"
export interface CreateNotice {
  type: typeof CREATE_NOTICE
  payload: INotice
}

export const CREATE_NOTICE_ERROR = "CREATE_NOTICE_ERROR"
export interface CreateNoticeError {
  type: typeof CREATE_NOTICE_ERROR
  payload: Error
}

export type NoticeActionTypes = CreateNotice | CreateNoticeError

// Auth actions
export const SIGN_UP = "SIGN_UP"
export interface SignUp {
  type: typeof SIGN_UP
}

export const SIGN_UP_ERROR = "SIGN_UP_ERROR"
export interface SignUpError {
  type: typeof SIGN_UP_ERROR
  payload: IFBError
}

export const MEMBER_PROFILE_CREATED = "MEMBER_PROFILE_CREATED"
export interface MemberProfileCreated {
  type: typeof MEMBER_PROFILE_CREATED
}

export const MEMBER_PROFILE_CREATED_ERROR = "MEMBER_PROFILE_CREATED_ERROR"
export interface MemberProfileCreatedError {
  type: typeof MEMBER_PROFILE_CREATED_ERROR
  payload: IFBError
}

export const SIGN_IN = "SIGN_IN"
export interface SignIn {
  type: typeof SIGN_IN
}

export const SIGN_IN_ERROR = "SIGN_IN_ERROR"
export interface SignInError {
  type: typeof SIGN_IN_ERROR
  payload: IFBError
}

export const REMEMBER_ME = "REMEMBER_ME"
export interface RememberMe {
  type: typeof REMEMBER_ME
  payload: boolean
}

export const REMEMBER_ME_ERROR = "REMEMBER_ME_ERROR"
export interface RememberMeError {
  type: typeof REMEMBER_ME_ERROR
  payload: IFBError
}

export const SIGN_OUT = "SIGN_OUT"
export interface SignOut {
  type: typeof SIGN_OUT
}

export const SIGN_OUT_ERROR = "SIGN_OUT_ERROR"
export interface SignOutError {
  type: typeof SIGN_OUT_ERROR
  payload: IFBError
}

export type AuthActionTypes =
  | SignUp
  | SignUpError
  | MemberProfileCreated
  | MemberProfileCreatedError
  | SignIn
  | SignInError
  | RememberMe
  | RememberMeError
  | SignOut
  | SignOutError

// All Action types
export type AppActions = NoticeActionTypes | AuthActionTypes
