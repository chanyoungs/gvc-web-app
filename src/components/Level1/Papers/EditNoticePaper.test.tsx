import React from 'react'
import Enzyme, { mount } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import { EditNoticePaper } from './EditNoticePaper'
import { INoticeWithMeta } from '../../../types'
import firebase from '../../../firebase/index'
import { useDispatch } from 'react-redux'

Enzyme.configure({
  adapter: new EnzymeAdapter(),
  disableLifecycleMethods: true
})

jest.mock('react-redux', () => ({
  useDispatch: jest.fn()
}))

const mockUseDispatch = useDispatch as jest.Mock
const updateNotice = jest.fn()
const mockDispatch = jest.fn(updateNotice)

describe('<EditNoticePaper />', () => {
  mockUseDispatch.mockReturnValue(mockDispatch)
  const notice: INoticeWithMeta = {
    content: 'content1',
    createdAt: firebase.firestore.Timestamp.now(),
    id: 'id1',
    title: 'title1'
  }

  const container = mount(<EditNoticePaper notice={notice} />)

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  test('renders without error', () => {
    expect(
      container.find('[data-testid="component-edit-noticepaper"]').length
    ).toEqual(1)
  })

  it('should have an title field', () => {
    expect(
      container.find('[data-testid="title"] input[type="text"]').length
    ).toEqual(1)
  })

  it('should have an content field', () => {
    expect(
      container.find('[data-testid="content"] textarea').length
    ).toBeGreaterThan(1)
  })

  it('should set the title value on change event', () => {
    container
      .find('[data-testid="title"] input[type="text"]')
      .simulate('change', {
        target: {
          value: 'some new title'
        }
      })

    expect(
      container.find('[data-testid="title"] input[type="text"]').prop('value')
    ).toBe('some new title')
  })

  it('should set the content value on change event', () => {
    container
      .find('[data-testid="content"] textarea')
      .at(0)
      .simulate('change', {
        target: {
          value: 'some new content'
        }
      })

    expect(
      container
        .find('[data-testid="content"] textarea')
        .at(0)
        .prop('value')
    ).toBe('some new content')
  })

  it('should call the add notice function on submit button click', () => {
    container
      .find('[data-testid="title"] input[type="text"]')
      .simulate('change', {
        target: {
          value: 'some new title'
        }
      })
    container
      .find('[data-testid="content"] textarea')
      .at(0)
      .simulate('change', {
        target: {
          value: 'some new content'
        }
      })

    container.find('button[type="button"]').simulate('click')

    expect(mockDispatch).toHaveBeenCalledTimes(1)

    expect(
      container
        .find('[data-testid="content"] textarea')
        .at(0)
        .prop('value')
    ).toBe('')

    expect(
      container.find('[data-testid="title"] input[type="text"]').prop('value')
    ).toBe('')
  })
})
