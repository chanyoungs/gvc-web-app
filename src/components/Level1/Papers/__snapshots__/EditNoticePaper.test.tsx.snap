// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditNoticePaper /> should match the snapshot 1`] = `"<div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div data-testid=\\"component-edit-noticepaper\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"title\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Title</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" type=\\"text\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"title1\\"></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"content\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"standard-multiline-flexible\\" id=\\"standard-multiline-flexible-label\\">Content</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline\\"><textarea rows=\\"1\\" aria-invalid=\\"false\\" id=\\"standard-multiline-flexible\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\" style=\\"height: 0px; overflow: hidden;\\">content1</textarea><textarea aria-hidden=\\"true\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;\\"></textarea></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit\\"><span class=\\"MuiButton-label\\">Update notice</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div>"`;
